<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Overview · GeneNetworkAPI.jl</title><script data-outdated-warner src="assets/warner.js"></script><link rel="canonical" href="https://senresearch.github.io/GeneNetworkAPI.jl/"/><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.045/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.13.24/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href>GeneNetworkAPI.jl</a></span></div><form class="docs-search" action="search/"><input class="docs-search-query" id="documenter-search-query" name="q" type="text" placeholder="Search docs"/></form><ul class="docs-menu"><li class="is-active"><a class="tocitem" href>Overview</a><ul class="internal"><li><a class="tocitem" href="#Credits"><span>Credits</span></a></li><li><a class="tocitem" href="#Note-on-terminology"><span>Note on terminology</span></a></li><li><a class="tocitem" href="#Check-connection"><span>Check connection</span></a></li><li><a class="tocitem" href="#Get-species-list"><span>Get species list</span></a></li><li><a class="tocitem" href="#List-groups-for-a-species"><span>List groups for a species</span></a></li><li><a class="tocitem" href="#Get-genotypes-for-a-group"><span>Get genotypes for a group</span></a></li><li><a class="tocitem" href="#List-datasets-for-a-group"><span>List datasets for a group</span></a></li><li><a class="tocitem" href="#List-meta-information-of-genotype-file"><span>List meta information of genotype file</span></a></li><li><a class="tocitem" href="#Get-sample-data-for-a-group"><span>Get sample data for a group</span></a></li><li><a class="tocitem" href="#Get-information-about-traits"><span>Get information about traits</span></a></li><li><a class="tocitem" href="#Summary-information-on-traits"><span>Summary information on traits</span></a></li><li><a class="tocitem" href="#Analysis-commands"><span>Analysis commands</span></a></li></ul></li><li><a class="tocitem" href="functions/">Functions</a></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Overview</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Overview</a></li></ul></nav><div class="docs-right"><a class="docs-edit-link" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/main/docs/src/index.md#" title="Edit on GitHub"><span class="docs-icon fab"></span><span class="docs-label is-hidden-touch">Edit on GitHub</span></a><a class="docs-settings-button fas fa-cog" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-sidebar-button fa fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a></div></header><article class="content" id="documenter-page"><h1 id="Overview"><a class="docs-heading-anchor" href="#Overview">Overview</a><a id="Overview-1"></a><a class="docs-heading-anchor-permalink" href="#Overview" title="Permalink"></a></h1><p>The <a href="https://github.com/sens/GeneNetworkAPI.jl">GeneNetworkAPI</a> package provides  access to the <a href="http://genenetwork.org">GeneNetwork</a> database and analysis functions using the <a href="https://github.com/genenetwork/gn-docs/blob/master/api/GN2-REST-API.md">GeneNetwork REST API</a>.</p><h2 id="Credits"><a class="docs-heading-anchor" href="#Credits">Credits</a><a id="Credits-1"></a><a class="docs-heading-anchor-permalink" href="#Credits" title="Permalink"></a></h2><p>Pjotr Prins and Zach Sloan are the main contributors to the GeneNetwork REST API.  Karl Broman wrote the <a href="https://github.com/kbroman/GNapi/blob/main/README.md">GNapi</a> R package for providing access to GeneNetwork from R.  This package follows the structure and function of that package closely.</p><h2 id="Note-on-terminology"><a class="docs-heading-anchor" href="#Note-on-terminology">Note on terminology</a><a id="Note-on-terminology-1"></a><a class="docs-heading-anchor-permalink" href="#Note-on-terminology" title="Permalink"></a></h2><p>GeneNetwork collects data on genetically segregating populations (called <em>groups</em>) in a number of <em>species</em> including humans.  Most of the phenotype data is &quot;omic&quot; data which are organized as <em>datasets</em>. </p><h2 id="Check-connection"><a class="docs-heading-anchor" href="#Check-connection">Check connection</a><a id="Check-connection-1"></a><a class="docs-heading-anchor-permalink" href="#Check-connection" title="Permalink"></a></h2><p>To check if the website is responding properly:</p><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; check_gn()</code><code class="nohighlight hljs ansi" style="display:block;">GeneNetwork is alive.
200</code></pre><h2 id="Get-species-list"><a class="docs-heading-anchor" href="#Get-species-list">Get species list</a><a id="Get-species-list-1"></a><a class="docs-heading-anchor-permalink" href="#Get-species-list" title="Permalink"></a></h2><p>Which species have data on them?</p><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; list_species()</code><code class="nohighlight hljs ansi" style="display:block;">12×4 DataFrame
 Row │ FullName                           Id     Name             TaxonomyId
     │ String                             Int64  String           Int64
─────┼───────────────────────────────────────────────────────────────────────
   1 │ Mus musculus                           1  mouse                 10090
   2 │ Rattus norvegicus                      2  rat                   10116
   3 │ Arabidopsis thaliana                   3  arabidopsis            3702
   4 │ Homo sapiens                           4  human                  9606
   5 │ Hordeum vulgare                        5  barley                 4513
   6 │ Fly (Drosophila melanogaster dm6)      6  drosophila             7227
   7 │ Macaca mulatta                         7  macaque monkey         9544
   8 │ Glycine max                            8  soybean                3847
   9 │ Solanum lycopersicum                   9  tomato                 4081
  10 │ Populus trichocarpa                   10  poplar                 3689
  11 │ Oryzias latipes (Japanese medaka)     11  Oryzias latipes        8090
  12 │ Bat (Glossophaga soricina)            12  bat                   27638</code></pre><p>To get information on a single species:</p><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; list_species(&quot;rat&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×4 DataFrame
 Row │ FullName           Id     Name    TaxonomyId
     │ String             Int64  String  Int64
─────┼──────────────────────────────────────────────
   1 │ Rattus norvegicus      2  rat          10116</code></pre><p>You could also subset (safer):</p><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; GeneNetworkAPI.subset(list_species(), :Name =&gt; x-&gt;x.==&quot;rat&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×4 DataFrame
 Row │ FullName           Id     Name    TaxonomyId
     │ String             Int64  String  Int64
─────┼──────────────────────────────────────────────
   1 │ Rattus norvegicus      2  rat          10116</code></pre><h2 id="List-groups-for-a-species"><a class="docs-heading-anchor" href="#List-groups-for-a-species">List groups for a species</a><a id="List-groups-for-a-species-1"></a><a class="docs-heading-anchor-permalink" href="#List-groups-for-a-species" title="Permalink"></a></h2><p>Since the information is organized by segregating population (&quot;group&quot;), it is useful to get a list for a preticular species you might be interested in.</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; list_groups(&quot;rat&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">7×8 DataFrame
 Row │ DisplayName                        FullName                           GeneticType  Id     MappingMethodId  Name             SpeciesId  public ⋯
     │ String                             String                             String       Int64  String           String           Int64      Int64  ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ Hybrid Rat Diversity Panel (Incl…  Hybrid Rat Diversity Panel (Incl…  None            10  1                HXBBXH                   2       2 ⋯
   2 │ UIOWA SRxSHRSP F2                  UIOWA SRxSHRSP F2                  intercross      24  1                SRxSHRSPF2               2       2
   3 │ NIH Heterogeneous Stock (RGSMC 2…  NIH Heterogeneous Stock (RGSMC 2…  None            42  1                HSNIH-RGSMC              2       2
   4 │ NIH Heterogeneous Stock (Palmer)   NIH Heterogeneous Stock (Palmer)   None            55  1                HSNIH-Palmer             2       2
   5 │ NWU WKYxF344 F2 Behavior           NWU WKYxF344 F2 Behavior           intercross      82  3                NWU_WKYxF344_F2          2       2 ⋯
   6 │ HIV-1Tg and Control                HIV-1Tg and Control                None            83  1                HIV-1Tg                  2       2
   7 │ HRDP-HXB/BXH Brain Proteome        HRDP-HXB/BXH Brain Proteome        None            87  1                HRDP_HXB-BXH-BP          2       2</code></pre><p>You can see the type of population it is.  Note the short name (<code>Name</code>) as that will be used in queries involving that population (group).</p><h2 id="Get-genotypes-for-a-group"><a class="docs-heading-anchor" href="#Get-genotypes-for-a-group">Get genotypes for a group</a><a id="Get-genotypes-for-a-group-1"></a><a class="docs-heading-anchor-permalink" href="#Get-genotypes-for-a-group" title="Permalink"></a></h2><p>To get the genotypes of a group:</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; get_geno(&quot;BXD&quot;) |&gt; (x-&gt;first(x,10))</code><code class="nohighlight hljs ansi" style="display:block;">10×239 DataFrame
 Row │ Chr      Locus           cM       Mb       BXD1     BXD2     BXD5     BXD6     BXD8     BXD9     BXD11    BXD12    BXD13    BXD14    BXD15    ⋯
     │ String3  String31        Float64  Float64  String1  String1  String1  String1  String1  String1  String1  String1  String1  String1  String1  ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ 1        rsm10000000001     0.0   3.00149  B        B        D        D        D        B        B        D        B        B        D        ⋯
   2 │ 1        rs31443144         0.11  3.01027  B        B        D        D        D        B        B        D        B        B        D
   3 │ 1        rs6269442          0.21  3.4922   B        B        D        D        D        B        B        D        B        B        D
   4 │ 1        rs32285189         0.32  3.5112   B        B        D        D        D        B        B        D        B        B        D
   5 │ 1        rs258367496        0.43  3.6598   B        B        D        D        D        B        B        D        B        B        D        ⋯
   6 │ 1        rs32430919         0.53  3.77702  B        B        D        D        D        B        B        D        B        B        D
   7 │ 1        rs36251697         0.64  3.81227  B        B        D        D        D        B        B        D        B        B        D
   8 │ 1        rs30658298         0.75  4.43062  B        B        D        D        D        B        B        D        B        B        D
   9 │ 1        rs31879829         0.85  4.51871  B        B        D        D        D        B        B        D        B        B        D        ⋯
  10 │ 1        rs36742481         0.96  4.77632  B        B        D        D        D        B        B        D        B        B        D
                                                                                                                                   224 columns omitted</code></pre><p>Currently, we only support the <code>.geno</code> format which returns a data frame of genotypes with rows as marker and columns as individuals.</p><h2 id="List-datasets-for-a-group"><a class="docs-heading-anchor" href="#List-datasets-for-a-group">List datasets for a group</a><a id="List-datasets-for-a-group-1"></a><a class="docs-heading-anchor-permalink" href="#List-datasets-for-a-group" title="Permalink"></a></h2><p>To list the (omic) datasets available for a group, you have to use the name as listed in the group list for a species:</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; list_datasets(&quot;HSNIH-Palmer&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">10×11 DataFrame
 Row │ AvgID  CreateTime                     DataScale  FullName                           Id     Long_Abbreviation              ProbeFreezeId  Shor ⋯
     │ Int64  String                         String     String                             Int64  String                         Int64          Stri ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │    24  Mon, 27 Aug 2018 00:00:00 GMT  log2       HSNIH-Palmer Nucleus Accumbens C…    860  HSNIH-Rat-Acbc-RSeq-Aug18                347  HSNI ⋯
   2 │    24  Sun, 26 Aug 2018 00:00:00 GMT  log2       HSNIH-Palmer Infralimbic Cortex …    861  HSNIH-Rat-IL-RSeq-Aug18                  348  HSNI
   3 │    24  Sat, 25 Aug 2018 00:00:00 GMT  log2       HSNIH-Palmer Lateral Habenula RN…    862  HSNIH-Rat-LHB-RSeq-Aug18                 349  HSNI
   4 │    24  Fri, 24 Aug 2018 00:00:00 GMT  log2       HSNIH-Palmer Prelimbic Cortex RN…    863  HSNIH-Rat-PL-RSeq-Aug18                  350  HSNI
   5 │    24  Thu, 23 Aug 2018 00:00:00 GMT  log2       HSNIH-Palmer Orbitofrontal Corte…    864  HSNIH-Rat-VoLo-RSeq-Aug18                351  HSNI ⋯
   6 │    24  Fri, 14 Sep 2018 00:00:00 GMT  log2       HSNIH-Palmer Nucleus Accumbens C…    868  HSNIH-Rat-Acbc-RSeqlog2-Aug18            347  HSNI
   7 │    24  Fri, 14 Sep 2018 00:00:00 GMT  log2       HSNIH-Palmer Infralimbic Cortex …    869  HSNIH-Rat-IL-RSeqlog2-Aug18              348  HSNI
   8 │    24  Fri, 14 Sep 2018 00:00:00 GMT  log2       HSNIH-Palmer Lateral Habenula RN…    870  HSNIH-Rat-LHB-RSeqlog2-Aug18             349  HSNI
   9 │    24  Fri, 14 Sep 2018 00:00:00 GMT  log2       HSNIH-Palmer Prelimbic Cortex RN…    871  HSNIH-Rat-PL-RSeqlog2-Aug18              350  HSNI ⋯
  10 │    24  Fri, 14 Sep 2018 00:00:00 GMT  log2       HSNIH-Palmer Orbitofrontal Corte…    872  HSNIH-Rat-VoLo-RSeqlog2-Aug18            351  HSNI
                                                                                                                                     4 columns omitted</code></pre><h2 id="List-meta-information-of-genotype-file"><a class="docs-heading-anchor" href="#List-meta-information-of-genotype-file">List meta information of genotype file</a><a id="List-meta-information-of-genotype-file-1"></a><a class="docs-heading-anchor-permalink" href="#List-meta-information-of-genotype-file" title="Permalink"></a></h2><p>To list the meta information of a genotype file for a group, you have to use the name as listed in the group list for a species:</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; list_geno(&quot;BXD&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">3×4 DataFrame
 Row │ f1s     mat       pat     location
     │ String  String    String  String
─────┼──────────────────────────────────────
   1 │ B6D2F1  C57BL/6J  DBA/2J  BXD.8.geno
   2 │ D2B6F1                    BXD.geno*
   3 │                           BXD.4.geno</code></pre><p>In the case where multiple genotype dataset are associated to a group,  the <code>*</code> indicates the default genotype dataset.</p><h2 id="Get-sample-data-for-a-group"><a class="docs-heading-anchor" href="#Get-sample-data-for-a-group">Get sample data for a group</a><a id="Get-sample-data-for-a-group-1"></a><a class="docs-heading-anchor-permalink" href="#Get-sample-data-for-a-group" title="Permalink"></a></h2><p>This gives you a matrix with rows as individuals/samples/strains and columns as &quot;clinical&quot; (non-omic) phenotypes.  The number after the underscore is the phenotype number (to be used later).  Some data may be missing.</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; get_pheno(&quot;HSNIH-Palmer&quot;) |&gt; (x-&gt;x[81:100,:]) |&gt; show</code><code class="nohighlight hljs ansi" style="display:block;">20×509 DataFrame
 Row │ id          HSR_10001   HSR_10002    HSR_10003   HSR_10004    HSR_10005  HSR_10006    HSR_10007  HSR_10008   HSR_10009   HSR_10010    HSR_100 ⋯
     │ String15    Float64?    Float64?     Float64?    Float64?     Float64?   Float64?     Float64?   Float64?    Float64?    Float64?     Float64 ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ 000721E489  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing ⋯
   2 │ 00072AAC0D  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
   3 │ 00072AC972  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
   4 │ 00077E61DC  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
   5 │ 00077E61EC  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing ⋯
   6 │ 00077E61F3  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
   7 │ 00077E61F5  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
   8 │ 00077E6204  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
  ⋮  │     ⋮           ⋮            ⋮           ⋮            ⋮           ⋮           ⋮           ⋮          ⋮           ⋮            ⋮           ⋮   ⋱
  14 │ 00077E634B  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing ⋯
  15 │ 00077E63D9  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
  16 │ 00077E641E  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
  17 │ 00077E6433  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
  18 │ 00077E64B3     8672.99       86.414     4762.08       63.416    24076.1       87.118       84.0       43.57     6614.97       22.526     1955 ⋯
  19 │ 00077E64BA  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
  20 │ 00077E64C1  missing     missing      missing     missing      missing    missing      missing    missing     missing     missing      missing
                                                                                                                        498 columns and 5 rows omitted</code></pre><h2 id="Get-information-about-traits"><a class="docs-heading-anchor" href="#Get-information-about-traits">Get information about traits</a><a id="Get-information-about-traits-1"></a><a class="docs-heading-anchor-permalink" href="#Get-information-about-traits" title="Permalink"></a></h2><p>To get information on a particular (non-omic) trait use the group name and the trait number:</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; info_dataset(&quot;HSNIH-Palmer&quot;,&quot;10308&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×4 DataFrame
 Row │ dataset_type  description                        id     name
     │ String        String                             Int64  String
─────┼───────────────────────────────────────────────────────────────────────────────
   1 │ phenotype     Central nervous system, behavior…  10308  reaction_time_pint1_5</code></pre><p>To get information on a dataset (of omic traits) for a group, use:</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; info_dataset(&quot;HSNIH-Rat-Acbc-RSeq-Aug18&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×10 DataFrame
 Row │ confidential  data_scale  dataset_type     full_name                          id     name                      public  short_name             ⋯
     │ Int64         String      String           String                             Int64  String                    Int64   String                 ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │            0  log2        mRNA expression  HSNIH-Palmer Nucleus Accumbens C…    860  HSNIH-Rat-Acbc-RSeq-0818       1  HSNIH-Palmer Nucleus A ⋯
                                                                                                                                     3 columns omitted</code></pre><h2 id="Summary-information-on-traits"><a class="docs-heading-anchor" href="#Summary-information-on-traits">Summary information on traits</a><a id="Summary-information-on-traits-1"></a><a class="docs-heading-anchor-permalink" href="#Summary-information-on-traits" title="Permalink"></a></h2><p>Get a list of the maximum LRS for each trait and position.</p><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; info_pheno(&quot;HXBBXH&quot;) |&gt; (x-&gt;first(x,10))</code><code class="nohighlight hljs ansi" style="display:block;">10×11 DataFrame
 Row │ Additive     Authors                            Chr     Description                         ⋯
     │ Float64?     String                             String  String                              ⋯
─────┼──────────────────────────────────────────────────────────────────────────────────────────────
   1 │  0.0499968   Pravenec M, Zidek V, Musilova A,…  8       Original post publication descri…   ⋯
   2 │ -0.0926364   Pravenec M, Zidek V, Musilova A,…  14      Original post publication descri…
   3 │  0.60189     Pravenec M, Zidek V, Musilova A,…  20      Original post publication descri…
   4 │ -0.543799    Pravenec M, Zidek V, Musilova A,…  5       Original post publication descri…
   5 │  0.00854221  Pravenec M, Zidek V, Musilova A,…  8       Original post publication descri…   ⋯
   6 │ -0.0142273   Pravenec M, Zidek V, Musilova A,…  14      Original post publication descri…
   7 │  0.427167    Pravenec M, Zidek V, Musilova A,…  1       Original post publication descri…
   8 │ -0.936806    Pravenec M, Zidek V, Musilova A,…  3       Original post publication descri…
   9 │ -0.635833    Pravenec M, Zidek V, Musilova A,…  3       Original post publication descri…   ⋯
  10 │ -0.681451    Pravenec M, Zidek V, Musilova A,…  7       Original post publication descri…
                                                                                   7 columns omitted</code></pre><p>You could also specify a group and a trait number or a dataset and a probename.</p><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; info_pheno(&quot;BXD&quot;,&quot;10001&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×4 DataFrame
 Row │ additive  id     locus       lrs
     │ Float64   Int64  String      Float64
─────┼──────────────────────────────────────
   1 │  2.39444      4  rs48756159  13.4975</code></pre><pre><code class="nohighlight hljs ansi" style="display:block;"></code><br/><code class="language-julia-repl hljs" style="display:block;">julia&gt; info_pheno(&quot;HC_M2_0606_P&quot;,&quot;1436869_at&quot;)</code><code class="nohighlight hljs ansi" style="display:block;">1×13 DataFrame
 Row │ additive   alias                              chr     description                id     locus      lrs      mb       mean     name        p_v ⋯
     │ Float64    String                             String  String                     Int64  String     Float64  Float64  Float64  String      Flo ⋯
─────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1 │ -0.214088  HHG1; HLP3; HPE3; SMMCI; Dsh; Hh…  5       sonic hedgehog (hedgehog)  99602  rs8253327  12.7711  28.4572  9.27909  1436869_at    0 ⋯
                                                                                                                                     3 columns omitted</code></pre><h2 id="Analysis-commands"><a class="docs-heading-anchor" href="#Analysis-commands">Analysis commands</a><a id="Analysis-commands-1"></a><a class="docs-heading-anchor-permalink" href="#Analysis-commands" title="Permalink"></a></h2><h3 id="GEMMA"><a class="docs-heading-anchor" href="#GEMMA">GEMMA</a><a id="GEMMA-1"></a><a class="docs-heading-anchor-permalink" href="#GEMMA" title="Permalink"></a></h3><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; run_gemma(&quot;BXDPublish&quot;,&quot;10015&quot;,use_loco=true) |&gt; (x-&gt;first(x,10))</code><code class="nohighlight hljs ansi" style="display:block;">10×6 DataFrame
 Row │ Mb       additive  chr  lod_score  name            p_value
     │ Float64  Float64   Any  Float64    String          Float64
─────┼─────────────────────────────────────────────────────────────
   1 │ 3.00149  0.497409  1     0.548648  rsm10000000001  0.282717
   2 │ 3.01027  0.497409  1     0.548648  rs31443144      0.282717
   3 │ 3.4922   0.497409  1     0.548648  rs6269442       0.282717
   4 │ 3.5112   0.497409  1     0.548648  rs32285189      0.282717
   5 │ 3.6598   0.497409  1     0.548648  rs258367496     0.282717
   6 │ 3.77702  0.497409  1     0.548648  rs32430919      0.282717
   7 │ 3.81227  0.497409  1     0.548648  rs36251697      0.282717
   8 │ 4.43062  0.497409  1     0.548648  rs30658298      0.282717
   9 │ 4.51871  0.497409  1     0.548648  rs31879829      0.282717
  10 │ 4.77632  0.497409  1     0.548648  rs36742481      0.282717</code></pre><h3 id="R/qtl"><a class="docs-heading-anchor" href="#R/qtl">R/qtl</a><a id="R/qtl-1"></a><a class="docs-heading-anchor-permalink" href="#R/qtl" title="Permalink"></a></h3><p>This function performs a one-dimensional genome scan.  The arguments are</p><ul><li>db (required) - DB name for trait above (Short_Abbreviation listed when you query for datasets)</li><li>trait (required) - ID for trait being mapped</li><li>method - hk (default) | ehk | em | imp | mr | mr-imp | mr-argmax ; Corresponds to the &quot;method&quot; option for the R/qtl scanone function.</li><li>model - normal (default) | binary | 2-part | np ; corresponds to the &quot;model&quot; option for the R/qtl scanone function</li><li>n_perm - number of permutations; 0 by default</li><li>control_marker - Name of marker to use as control; this relies on the user knowing the name of the marker they want to use as a covariate</li><li>interval_mapping - Whether to use interval mapping; &quot;false&quot; by default</li></ul><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; run_rqtl(&quot;BXDPublish&quot;, &quot;10015&quot;) |&gt; (x-&gt;first(x,10))</code><code class="nohighlight hljs ansi" style="display:block;">http://gn2.genenetwork.org/api/v_pre1/mapping?trait_id=10015&amp;db=BXDPublish&amp;method=rqtl&amp;rqtl_method=hk&amp;rqtl_model=normal&amp;num_perm=0&amp;interval_mapping=false
10×5 DataFrame
 Row │ Mb       cM       chr  lod_score  name
     │ Float64  Float64  Any  Float64    String
─────┼───────────────────────────────────────────────
   1 │ 3.01027  3.01027  1     0.116927  rs31443144
   2 │ 3.4922   3.4922   1     0.117404  rs6269442
   3 │ 3.5112   3.5112   1     0.117424  rs32285189
   4 │ 3.6598   3.6598   1     0.117573  rs258367496
   5 │ 3.77702  3.77702  1     0.117691  rs32430919
   6 │ 3.81227  3.81227  1     0.117727  rs36251697
   7 │ 4.43062  4.43062  1     0.118356  rs30658298
   8 │ 4.44674  4.44674  1     0.118372  rs51852623
   9 │ 4.51871  4.51871  1     0.118447  rs31879829
  10 │ 4.77632  4.77632  1     0.118714  rs36742481</code></pre><h3 id="Correlation"><a class="docs-heading-anchor" href="#Correlation">Correlation</a><a id="Correlation-1"></a><a class="docs-heading-anchor-permalink" href="#Correlation" title="Permalink"></a></h3><p>This function correlates a trait in a dataset against all traits in a target database.</p><ul><li>trait_id (required) - ID for trait used for correlation</li><li>db (required) - DB name for the trait above (this is the Short_Abbreviation listed when you query for datasets)</li><li>target_db (required) - Target DB name to be correlated against</li><li>type - sample (default) | tissue</li><li>method - pearson (default) | spearman</li><li>return - Number of results to return (default = 500)</li></ul><pre><code class="language-julia-repl hljs" style="display:block;">julia&gt; run_correlation(&quot;1427571_at&quot;,&quot;HC_M2_0606_P&quot;,&quot;BXDPublish&quot;) |&gt; (x-&gt;first(x,10))</code><code class="nohighlight hljs ansi" style="display:block;">10×4 DataFrame
 Row │ #_strains  p_value      sample_r   trait
     │ Int64      Float64      Float64    String
─────┼───────────────────────────────────────────
   1 │         6  0.00480466   -0.942857  20511
   2 │         6  0.00480466   -0.942857  20724
   3 │        12  1.82889e-5   -0.923362  13536
   4 │         7  0.00680719    0.892857  10157
   5 │         7  0.00680719   -0.892857  20392
   6 │         6  0.0188455     0.885714  20479
   7 │        12  0.000189298  -0.875658  12762
   8 │        12  0.000245942   0.868653  12760
   9 │         7  0.0136973    -0.857143  20559
  10 │        10  0.00222003   -0.842424  10925</code></pre></article><nav class="docs-footer"><a class="docs-footer-nextpage" href="functions/">Functions »</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 0.27.24 on <span class="colophon-date" title="Wednesday 12 April 2023 02:23">Wednesday 12 April 2023</span>. Using Julia version 1.8.5.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
