<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Functions · GeneNetworkAPI.jl</title><meta name="title" content="Functions · GeneNetworkAPI.jl"/><meta property="og:title" content="Functions · GeneNetworkAPI.jl"/><meta property="twitter:title" content="Functions · GeneNetworkAPI.jl"/><meta name="description" content="Documentation for GeneNetworkAPI.jl."/><meta property="og:description" content="Documentation for GeneNetworkAPI.jl."/><meta property="twitter:description" content="Documentation for GeneNetworkAPI.jl."/><meta property="og:url" content="https://senresearch.github.io/GeneNetworkAPI.jl/functions/"/><meta property="twitter:url" content="https://senresearch.github.io/GeneNetworkAPI.jl/functions/"/><link rel="canonical" href="https://senresearch.github.io/GeneNetworkAPI.jl/functions/"/><script data-outdated-warner src="../assets/warner.js"></script><link href="https://cdnjs.cloudflare.com/ajax/libs/lato-font/3.0.0/css/lato-font.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/juliamono/0.050/juliamono.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/fontawesome.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/solid.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/brands.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.16.8/katex.min.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL=".."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" data-main="../assets/documenter.js"></script><script src="../search_index.js"></script><script src="../siteinfo.js"></script><script src="../../versions.js"></script><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-dark.css" data-theme-name="documenter-dark" data-theme-primary-dark/><link class="docs-theme-link" rel="stylesheet" type="text/css" href="../assets/themes/documenter-light.css" data-theme-name="documenter-light" data-theme-primary/><script src="../assets/themeswap.js"></script></head><body><div id="documenter"><nav class="docs-sidebar"><div class="docs-package-name"><span class="docs-autofit"><a href="../">GeneNetworkAPI.jl</a></span></div><button class="docs-search-query input is-rounded is-small is-clickable my-2 mx-auto py-1 px-2" id="documenter-search-query">Search docs (Ctrl + /)</button><ul class="docs-menu"><li><a class="tocitem" href="../">Overview</a></li><li class="is-active"><a class="tocitem" href>Functions</a><ul class="internal"><li><a class="tocitem" href="#Server-functions"><span>Server functions</span></a></li><li><a class="tocitem" href="#List-what&#39;s-in-the-database"><span>List what&#39;s in the database</span></a></li><li><a class="tocitem" href="#Get-data-and-information"><span>Get data and information</span></a></li><li><a class="tocitem" href="#Download-data"><span>Download data</span></a></li><li><a class="tocitem" href="#Display-table"><span>Display table</span></a></li><li><a class="tocitem" href="#Utils"><span>Utils</span></a></li><li><a class="tocitem" href="#Run-jobs-on-the-server"><span>Run jobs on the server</span></a></li></ul></li></ul><div class="docs-version-selector field has-addons"><div class="control"><span class="docs-label button is-static is-size-7">Version</span></div><div class="docs-selector control is-expanded"><div class="select is-fullwidth is-size-7"><select id="documenter-version-selector"></select></div></div></div></nav><div class="docs-main"><header class="docs-navbar"><a class="docs-sidebar-button docs-navbar-link fa-solid fa-bars is-hidden-desktop" id="documenter-sidebar-button" href="#"></a><nav class="breadcrumb"><ul class="is-hidden-mobile"><li class="is-active"><a href>Functions</a></li></ul><ul class="is-hidden-tablet"><li class="is-active"><a href>Functions</a></li></ul></nav><div class="docs-right"><a class="docs-navbar-link" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/main/docs/src/functions.md#" title="Edit source on GitHub"><span class="docs-icon fa-solid"></span></a><a class="docs-settings-button docs-navbar-link fa-solid fa-gear" id="documenter-settings-button" href="#" title="Settings"></a><a class="docs-article-toggle-button fa-solid fa-chevron-up" id="documenter-article-toggle-button" href="javascript:;" title="Collapse all docstrings"></a></div></header><article class="content" id="documenter-page"><h1 id="Functions"><a class="docs-heading-anchor" href="#Functions">Functions</a><a id="Functions-1"></a><a class="docs-heading-anchor-permalink" href="#Functions" title="Permalink"></a></h1><ul><li><a href="#GeneNetworkAPI.check_gn"><code>GeneNetworkAPI.check_gn</code></a></li><li><a href="#GeneNetworkAPI.download_geno"><code>GeneNetworkAPI.download_geno</code></a></li><li><a href="#GeneNetworkAPI.download_omics"><code>GeneNetworkAPI.download_omics</code></a></li><li><a href="#GeneNetworkAPI.download_pheno"><code>GeneNetworkAPI.download_pheno</code></a></li><li><a href="#GeneNetworkAPI.genofile_location"><code>GeneNetworkAPI.genofile_location</code></a></li><li><a href="#GeneNetworkAPI.get_geno"><code>GeneNetworkAPI.get_geno</code></a></li><li><a href="#GeneNetworkAPI.get_omics"><code>GeneNetworkAPI.get_omics</code></a></li><li><a href="#GeneNetworkAPI.get_pheno"><code>GeneNetworkAPI.get_pheno</code></a></li><li><a href="#GeneNetworkAPI.gn_url"><code>GeneNetworkAPI.gn_url</code></a></li><li><a href="#GeneNetworkAPI.has_error_500"><code>GeneNetworkAPI.has_error_500</code></a></li><li><a href="#GeneNetworkAPI.has_genofile_meta"><code>GeneNetworkAPI.has_genofile_meta</code></a></li><li><a href="#GeneNetworkAPI.info_dataset"><code>GeneNetworkAPI.info_dataset</code></a></li><li><a href="#GeneNetworkAPI.info_pheno"><code>GeneNetworkAPI.info_pheno</code></a></li><li><a href="#GeneNetworkAPI.list_datasets"><code>GeneNetworkAPI.list_datasets</code></a></li><li><a href="#GeneNetworkAPI.list_geno"><code>GeneNetworkAPI.list_geno</code></a></li><li><a href="#GeneNetworkAPI.list_groups"><code>GeneNetworkAPI.list_groups</code></a></li><li><a href="#GeneNetworkAPI.list_species"><code>GeneNetworkAPI.list_species</code></a></li><li><a href="#GeneNetworkAPI.make_rectangular"><code>GeneNetworkAPI.make_rectangular</code></a></li><li><a href="#GeneNetworkAPI.parse_geno"><code>GeneNetworkAPI.parse_geno</code></a></li><li><a href="#GeneNetworkAPI.run_correlation"><code>GeneNetworkAPI.run_correlation</code></a></li><li><a href="#GeneNetworkAPI.run_gemma"><code>GeneNetworkAPI.run_gemma</code></a></li><li><a href="#GeneNetworkAPI.run_rqtl"><code>GeneNetworkAPI.run_rqtl</code></a></li><li><a href="#GeneNetworkAPI.show_table"><code>GeneNetworkAPI.show_table</code></a></li></ul><h2 id="Server-functions"><a class="docs-heading-anchor" href="#Server-functions">Server functions</a><a id="Server-functions-1"></a><a class="docs-heading-anchor-permalink" href="#Server-functions" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.gn_url" href="#GeneNetworkAPI.gn_url"><code>GeneNetworkAPI.gn_url</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">gn_url()</code></pre><p>Return the default GeneNetwork server API URL.</p><p>All user-level functions optionally take a GeneNetwork server URL that could be different from the standard URL in case one wanted to query a different instance of the server.</p><p><strong>Example</strong></p><pre><code class="language-julia-repl hljs">julia&gt; gn_url()
&quot;https://genenetwork.org/api/v_pre1/&quot;</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/query.jl#L5-L20">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.check_gn" href="#GeneNetworkAPI.check_gn"><code>GeneNetworkAPI.check_gn</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">check_gn(url::AbstractString)</code></pre><p>Check if GeneNetwork server is responding properly.</p><p>Returns the HTTP status code (<code>200</code> if successful) and prints a message.</p><p><strong>Example</strong></p><pre><code class="language-julia-repl hljs">julia&gt; check_gn()
GeneNetwork is alive.
200</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/query.jl#L25-L39">source</a></section></article><h2 id="List-what&#39;s-in-the-database"><a class="docs-heading-anchor" href="#List-what&#39;s-in-the-database">List what&#39;s in the database</a><a id="List-what&#39;s-in-the-database-1"></a><a class="docs-heading-anchor-permalink" href="#List-what&#39;s-in-the-database" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.list_datasets" href="#GeneNetworkAPI.list_datasets"><code>GeneNetworkAPI.list_datasets</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">list_datasets(group::String=&quot;&quot;;gn_url::String=gn_url())</code></pre><p>Lists all datasets available in a specified <code>group</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/list_data.jl#L38-L43">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.list_groups" href="#GeneNetworkAPI.list_groups"><code>GeneNetworkAPI.list_groups</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">list_groups(species::String=&quot;&quot;;gn_url::String=gn_url())</code></pre><p>If <code>species</code> is not specified, then it returns all groups (segregating populations) represented in the GeneNetwork database.  If the string <code>species</code> is specified, then it returns all groups for that species.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/list_data.jl#L22-L29">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.list_species" href="#GeneNetworkAPI.list_species"><code>GeneNetworkAPI.list_species</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">list_species(species::String=&quot;&quot;;gn_url::String=gn_url())</code></pre><p>Returns a data frame with a list of species respresented in the GeneNetwork database.  If the <code>species</code> string is non-empty, it will return the information of the matched species.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/list_data.jl#L6-L13">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.list_geno" href="#GeneNetworkAPI.list_geno"><code>GeneNetworkAPI.list_geno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">list_geno(group::String; gn_url::String=gn_url())</code></pre><p>Returns a data frame with the location name of the different geno files of  a group, and if available some metadata such as strain of the first filial generation, maternal and paternal strain.  If there exist more than one location, the default location is indicated  by a <code>*</code>. </p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/list_data.jl#L48-L56">source</a></section></article><h2 id="Get-data-and-information"><a class="docs-heading-anchor" href="#Get-data-and-information">Get data and information</a><a id="Get-data-and-information-1"></a><a class="docs-heading-anchor-permalink" href="#Get-data-and-information" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.get_geno" href="#GeneNetworkAPI.get_geno"><code>GeneNetworkAPI.get_geno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_geno(group::String,format::String=&quot;geno&quot;;gn_url::String=gn_url())</code></pre><p>Returns the genotype matrix for a <code>group</code> in a given <code>format</code>.</p><p>Currently works only for files in the <code>geno</code> format.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/get_data.jl#L11-L18">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.get_pheno" href="#GeneNetworkAPI.get_pheno"><code>GeneNetworkAPI.get_pheno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_pheno(dataset::String;gn_url::String=gn_url())</code></pre><p>Returns the non-omic (&quot;clinical&quot;) phenotypes for a given <code>dataset</code>.</p><hr/><pre><code class="nohighlight hljs">get_pheno(dataset::String,trait::String; gn_url::String=gn_url())</code></pre><p>Returns a given <code>trait</code> in a <code>group</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/get_data.jl#L31-L42">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.get_omics" href="#GeneNetworkAPI.get_omics"><code>GeneNetworkAPI.get_omics</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">get_omics(dataset::String;gn_url::String=gn_url())</code></pre><p>Returns the omic phenotypes for a given <code>dataset</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/get_data.jl#L57-L62">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.info_dataset" href="#GeneNetworkAPI.info_dataset"><code>GeneNetworkAPI.info_dataset</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">info_dataset(dataset::String, trait::String=&quot;&quot;; gn_url::String = gn_url())</code></pre><p>Returns information about a <code>dataset</code> (if <code>trait</code> is empty), or else a given <code>trait</code> in a specified <code>dataset</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/info_data.jl#L6-L11">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.info_pheno" href="#GeneNetworkAPI.info_pheno"><code>GeneNetworkAPI.info_pheno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">info_pheno(group::String; gn_url::String=gn_url())</code></pre><p>Returns the maximum LRS for each <code>trait</code> in a <code>group</code>`.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/info_data.jl#L21-L25">source</a></section><section><div><pre><code class="language-julia hljs">info_pheno(group::String, trait::String; gn_url::String=gn_url())</code></pre><p>Returns the maximum LRS for each <code>trait</code> in a <code>group</code>, or alternatively a probe (use <code>trait</code>) in dataset (use <code>group</code>).</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/info_data.jl#L31-L36">source</a></section></article><h2 id="Download-data"><a class="docs-heading-anchor" href="#Download-data">Download data</a><a id="Download-data-1"></a><a class="docs-heading-anchor-permalink" href="#Download-data" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.download_geno" href="#GeneNetworkAPI.download_geno"><code>GeneNetworkAPI.download_geno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_geno(group::String,format::String=&quot;geno&quot;;
                gn_url::String=gn_url(),
                path::String = &quot;&quot;)</code></pre><p>Download the genotype matrix for a <code>group</code> in a given <code>format</code>.</p><p>Currently works only for files in the <code>geno</code> format.</p><p>The file will be downloaded in <code>path</code> location, if  <code>path</code> is not specified, then  the file will be a temporary file.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/download_data.jl#L10-L23">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.download_pheno" href="#GeneNetworkAPI.download_pheno"><code>GeneNetworkAPI.download_pheno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_pheno(dataset::String;gn_url::String=gn_url())</code></pre><p>Downloads the non-omic (&quot;clinical&quot;) phenotypes for a given <code>dataset</code>. The file will be downloaded in <code>path</code> location, if  <code>path</code> is not specified, then  the file will be a temporary file.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/download_data.jl#L52-L59">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.download_omics" href="#GeneNetworkAPI.download_omics"><code>GeneNetworkAPI.download_omics</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">download_omics(dataset::String; 
                gn_url::String=gn_url(), 
                path::String = tempname())</code></pre><p>Download the omic phenotypes for a given <code>dataset</code>. The file will be downloaded in <code>path</code> location, if  <code>path</code> is not specified, then  the file will be a temporary file.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/download_data.jl#L71-L80">source</a></section></article><h2 id="Display-table"><a class="docs-heading-anchor" href="#Display-table">Display table</a><a id="Display-table-1"></a><a class="docs-heading-anchor-permalink" href="#Display-table" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.show_table" href="#GeneNetworkAPI.show_table"><code>GeneNetworkAPI.show_table</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">show_table(df::DataFrame)</code></pre><p>Displays a dataframe in <code>pretty</code> mode.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/show_data.jl#L2-L6">source</a></section></article><h2 id="Utils"><a class="docs-heading-anchor" href="#Utils">Utils</a><a id="Utils-1"></a><a class="docs-heading-anchor-permalink" href="#Utils" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.make_rectangular" href="#GeneNetworkAPI.make_rectangular"><code>GeneNetworkAPI.make_rectangular</code></a> — <span class="docstring-category">Function</span></header><section><div><p>make<em>rectangular(v</em>arrays::Vector{Vector{T}} where T)</p><p>Takes an array of arrays of any type and make it a matrix of <code>String</code>.     Arrays can be of different type and length. To make the data rectangular,      it uses the <code>filler</code> argument, by default <code>filler</code> is an empty string.</p><p><strong>Example 1</strong></p><pre><code class="language-julia hljs">julia&gt; myArrayOfArrays = [[&quot;A1&quot;,&quot;A2&quot;, &quot;A3&quot;], [&quot;B1&quot;, &quot;B2&quot;], [&quot;C1&quot;,&quot;C2&quot;]];
julia&gt; make_rectangular(test)
3×3 Matrix{String}:
&quot;A1&quot;  &quot;B1&quot;  &quot;C1&quot;
&quot;A2&quot;  &quot;B2&quot;  &quot;C2&quot;
&quot;A3&quot;  &quot;&quot;    &quot;&quot;</code></pre><p><strong>Example 2</strong></p><pre><code class="language-julia hljs">julia&gt; myArrayOfArrays = [[&quot;A1&quot;,&quot;A2&quot;, &quot;A3&quot;], [1, 2], [&quot;C1&quot;,&quot;C2&quot;]];
julia&gt; make_rectangular(test)
3×3 Matrix{String}:
&quot;A1&quot;  &quot;1&quot;  &quot;C1&quot;
&quot;A2&quot;  &quot;2&quot;  &quot;C2&quot;
&quot;A3&quot;  &quot;&quot;    &quot;&quot;</code></pre></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/utils.jl#L3-L29">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.has_error_500" href="#GeneNetworkAPI.has_error_500"><code>GeneNetworkAPI.has_error_500</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">has_error_500(s::String)</code></pre><p>Returns <code>true</code> if the server encountered an internal error and was  unable to complete the request, otherwise it returns <code>false</code>.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/utils.jl#L43-L49">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.parse_geno" href="#GeneNetworkAPI.parse_geno"><code>GeneNetworkAPI.parse_geno</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">parse_geno(filename::String)</code></pre><p>Function to parse a .geno file, and returns the genotype dataframe.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/utils_geno.jl#L2-L8">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.genofile_location" href="#GeneNetworkAPI.genofile_location"><code>GeneNetworkAPI.genofile_location</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">genofile_location(json_parsed::Dict)</code></pre><p>Returns a vector of String containing the location of geno files for a group.</p><hr/><pre><code class="nohighlight hljs">genofile_location(group::String)</code></pre><p>Returns a vector of String containing the location of geno files for a group.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/utils_geno.jl#L24-L35">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.has_genofile_meta" href="#GeneNetworkAPI.has_genofile_meta"><code>GeneNetworkAPI.has_genofile_meta</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">has_genofile_meta(group::String; gn_url::String=gn_url())</code></pre><p>Returns <code>true</code> if the group&#39;s genotype files possesse metadata that may contain  alternative location of the files.</p></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/utils_geno.jl#L66-L72">source</a></section></article><h2 id="Run-jobs-on-the-server"><a class="docs-heading-anchor" href="#Run-jobs-on-the-server">Run jobs on the server</a><a id="Run-jobs-on-the-server-1"></a><a class="docs-heading-anchor-permalink" href="#Run-jobs-on-the-server" title="Permalink"></a></h2><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.run_correlation" href="#GeneNetworkAPI.run_correlation"><code>GeneNetworkAPI.run_correlation</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">run_correlation(trait::String, dataset::String, group::String; tp:;String=&quot;sample&quot;,
                method::String=&quot;pearson&quot;, n_result::Int64=500,
                gn_url::String=gn_url())</code></pre><p>This function correlates a <code>trait</code> in a <code>dataset</code> against all traits in a target database (<code>group</code>).</p><p>This query currently takes the following parameters:</p><ul><li>trait (required): trait ID for which the correlation is wanted</li><li>dataset (required): dataset in which the trait occurs (use the short abbreviation)</li><li>group (required): group in which which the comparisons will be performed</li><li>tp: sample (default) | tissue</li><li>method: pearson (default) | spearman</li><li>n_result: maximum number of results to return (default = 500)</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/run_data.jl#L77-L93">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.run_gemma" href="#GeneNetworkAPI.run_gemma"><code>GeneNetworkAPI.run_gemma</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">run_gemma(dataset::String,trait::String; use_loco::Bool=false, maf::Float64=0.01,
    gn_url::String=gn_url())</code></pre><p>Runs GEMMA with the specified <code>trait</code> in the specified <code>dataset</code>. Optional arguments include:</p><ul><li>use_loco (default=false): if LOCO (leave one chromosome out) should be used</li><li>maf (default=0.01): the minimum minor allele frequency of the markers to be scanned</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/run_data.jl#L7-L16">source</a></section></article><article class="docstring"><header><a class="docstring-article-toggle-button fa-solid fa-chevron-down" href="javascript:;" title="Collapse docstring"></a><a class="docstring-binding" id="GeneNetworkAPI.run_rqtl" href="#GeneNetworkAPI.run_rqtl"><code>GeneNetworkAPI.run_rqtl</code></a> — <span class="docstring-category">Function</span></header><section><div><pre><code class="language-julia hljs">function run_rqtl(dataset::String, trait::String; method::String=&quot;hk&quot;,
              model::String=&quot;normal&quot;, n_perm::Int64=0,
              control_marker::String=&quot;&quot;,
              gn_url::String=gn_url())</code></pre><p>Runs R/qtl&#39;s <code>scanone</code> function with the specified <code>trait</code> in the specified <code>dataset</code>.</p><p>Some optional arguments have analogs in options in the parent <code>scanone</code> function.  See the <a href="https://rqtl.org/manual/qtl-manual.pdf">help for that function</a>.</p><p>Optional arguments inlcude:</p><ul><li>method (default=&quot;hk&quot;): possible values are<ul><li>hk: Haley-Knott method</li><li>em: EM algorithm</li><li>ehk: Extended EM algorithm</li><li>imp: multiple imputation</li><li>mr: marker regression dropping individuals with missing genotypes</li><li>mr-imp: marker regression filling in missing data with a single imputation</li><li>mr-argmax: marker regression filling in by the Viterbi algorithm</li></ul></li><li>model (default=&quot;normal&quot;): model for trait given genotypes; possible values are<ul><li>normal: equivalent to linear regression with complete data</li><li>binary: equivalent to logistic regression with complete data</li><li>2part: two-part model for trait given genotype</li><li>np: non-parametric analysis</li></ul></li><li>n_perm (default=0) number of permutations to perform</li><li>control_marker: name of marker to control for as an additive covariate</li></ul></div><a class="docs-sourcelink" target="_blank" href="https://github.com/senresearch/GeneNetworkAPI.jl/blob/7e587eee2048ee35d9230cf2e7bf81d63fd43075/src/run_data.jl#L25-L53">source</a></section></article></article><nav class="docs-footer"><a class="docs-footer-prevpage" href="../">« Overview</a><div class="flexbox-break"></div><p class="footer-message">Powered by <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> and the <a href="https://julialang.org/">Julia Programming Language</a>.</p></nav></div><div class="modal" id="documenter-settings"><div class="modal-background"></div><div class="modal-card"><header class="modal-card-head"><p class="modal-card-title">Settings</p><button class="delete"></button></header><section class="modal-card-body"><p><label class="label">Theme</label><div class="select"><select id="documenter-themepicker"><option value="documenter-light">documenter-light</option><option value="documenter-dark">documenter-dark</option><option value="auto">Automatic (OS)</option></select></div></p><hr/><p>This document was generated with <a href="https://github.com/JuliaDocs/Documenter.jl">Documenter.jl</a> version 1.1.2 on <span class="colophon-date" title="Tuesday 31 October 2023 19:25">Tuesday 31 October 2023</span>. Using Julia version 1.9.3.</p></section><footer class="modal-card-foot"></footer></div></div></div></body></html>
